src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -395,6 +396,28 @@ trait UpdateSchema {
   val updateSchema: StructType
 }
 
-abstract class DeleteSchema {
-  val deleteSchema: StructType = StructType(Seq(StructField("id", DataTypes.LongType)))
+trait UpdateSchemaFromUpsertSchema extends UpsertSchema with UpdateSchema {
+  override val updateSchema: StructType = upsertSchema
 }
+
+trait ReadSchema {
+  val readSchema: StructType
+}
+
+trait DeleteSchema {
+  val deleteSchema: StructType
+}
+
+trait DeleteWithIdSchema extends DeleteSchema {
+  override val deleteSchema: StructType = StructType(Seq(StructField("id", DataTypes.LongType)))
+}
+
+trait DeleteWithExternalIdSchema extends DeleteSchema {
+  import cognite.spark.compiletime.macros.StructTypeEncoderMacro._
+
+  override val deleteSchema: StructType = structType[DeleteSchemaWithExternalId]()
+}
+
+final case class DeleteSchemaWithExternalId(
```

Comment:
```suggestion
final case class DeleteByExternalId(
```
or some other name, DeleteWithExternalIdSchema vs DeleteSchemaWithExternalId look too similar yet are slightly different things